test/unit/conditions/base/condition.test.ts

```diff
@@ -11,37 +11,45 @@ import {
 } from '../../testVariables';
 
 describe('validation', () => {
-  // TODO: Consider:
-  //   Use Condition here with returnTestValue schema
-  //   Refactor returnTestValue to be the part of the Condition
-  const condition = new ERC721Balance();
+  const condition = new ERC721Balance({
+    contractAddress: TEST_CONTRACT_ADDR,
+    chain: TEST_CHAIN_ID,
+  });
 
   it('accepts a correct schema', async () => {
-    const result = condition.validate({
-      contractAddress: TEST_CONTRACT_ADDR,
-      chain: TEST_CHAIN_ID,
-    });
+    const result = condition.validate();
     expect(result.error).toBeUndefined();
-    expect(result.value.contractAddress).toEqual(TEST_CONTRACT_ADDR);
+    expect(result.data.contractAddress).toEqual(TEST_CONTRACT_ADDR);
   });
 
-  it('updates on a valid schema value', async () => {
-    const result = condition.validate({
+  it('accepts on a valid value override', async () => {
+    const validOverride = {
       chain: TEST_CHAIN_ID,
       contractAddress: TEST_CONTRACT_ADDR_2,
-    });
+    };
+    const result = condition.validate(validOverride);
     expect(result.error).toBeUndefined();
-    expect(result.value.chain).toEqual(TEST_CHAIN_ID);
+    expect(result.data).toMatchObject(validOverride);
   });
 
-  it('rejects on an invalid schema value', async () => {
-    const result = condition.validate({
+  it('rejects on an invalid value override', async () => {
+    const invalidOverride = {
       chain: -1,
       contractAddress: TEST_CONTRACT_ADDR,
+    };
+    const result = condition.validate(invalidOverride);
+    expect(result.error).toBeDefined();
+    expect(result.data).toBeUndefined();
+    expect(result.error?.format()).toMatchObject({
+      chain: {
+        _errors: [
+          'Invalid literal value, expected 137',
+          'Invalid literal value, expected 80001',
+          'Invalid literal value, expected 5',
+          'Invalid literal value, expected 1',
```

Comment:
All of these are included in the error? I guess zod does do something like an array of literals so that the error would be like `"Invalid literal value, not one of 137, 80001, 5, 1"`

Reply:
I'm afraid this is what Zod does when you use `z.literal`. I've been looking for a way around that without much success, but I can give it another try.

Reply:
In my original comment, I meant to say *"I guess zod does [not] do something..."*

If you've already looked and it's not obvious then no worries. Thanks!